Updated and fixed ion.rangeSlider types.

Karel-van-de-Plassche opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If removing a declaration:
- [x] Delete the package's directory.

rbuckton commented on types/ion-rangeslider/index.d.ts:
```diff
@@ -0,0 +1,80 @@
+// Type definitions for ion-rangeslider 11.1
```
The highest published package version for ion-rangeslider is 2.2, not 11.1.

rbuckton replied:
I take it the path change from `ion.rangeSlider` to `ion-rangeslider` is to reflect the published package name in NPM?

rbuckton commented on types/ion-rangeslider/util/index_template.d.ts:
```diff
@@ -0,0 +1,25 @@
+// Type definitions for ion-rangeslider 11.1
```
What is the purpose of this file?

rbuckton commented on types/ion-rangeslider/util/parse_settings.py:
```diff
@@ -0,0 +1,155 @@
+from IPython import embed
```
What is the purpose of adding this file to DefinitelyTyped?

Karel-van-de-Plassche replied:
Automatically parse and create types directly from ion.rangeSlider documentation. I'd be open to remove this from DefinitelyTyped, and instead save this in another repository. If so, would I can add a link in a comment instead.

rbuckton commented on types/ion-rangeslider/util/.gitignore:
```diff
@@ -0,0 +1 @@
+index.d.ts
```
What is the purpose of adding this file to DefinitelyTyped?

rbuckton commented on types/ion-rangeslider/ion-rangeslider-tests.ts:
```diff
@@ -0,0 +1 @@
+import "ion-rangeslider";
```
The previous version had a test file with significantly more content. Please add or restore the tests.

rbuckton commented on types/ion.rangeslider/v1/index.d.ts:
```diff
@@ -1,52 +0,0 @@
-// Type definitions for for Ion.RangeSlider 1.9.1
```
These major-version-specific files should probably be moved over to the new folder.

rbuckton commented on types/ion-rangeslider/UNUSED_FILES.txt:
```diff
@@ -0,0 +1,3 @@
+util/index_template.d.ts
```
What is the purpose of adding this file to DefinitelyTyped?

Karel-van-de-Plassche replied:
Avoids failing of tests (see comment `types/ion-rangeslider/util/parse_settings.py`)

Karel-van-de-Plassche commented:
@rbuckton Hey! Any updates? Do you think I should remove the auto-parsing files to get this accepted?

Karel-van-de-Plassche commented:
@RyanCavanaugh Hey! It seems typescript-bot doesn't know what to do in this case. I've addressed the comments of the reviewer, so I think the label should be 'Awaiting reviewer feedback' and not 'revision needed'.

minestarks commented:
@Karel-van-de-Plassche Sorry for the delay. Please go ahead and remove the unused files. Once you do that I'll merge.

Karel-van-de-Plassche commented:
@minestarks No worries, thanks! In the meantime, I had made another pull request here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26312. It has a cleaner commit history (not sure what the merge policy is for DT) If you could merge that one, great, if not, I'll remove the files from this one tomorrow.

minestarks commented:
Cool, does #26312 supersede this one then? I will close this one if that's the case.

Karel-van-de-Plassche commented:
Yes, does. Feel free to close this one

Karel-van-de-Plassche commented:
Superseded by #26312 (merged)
